castor-service/src/test/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingServiceTest.java

```diff
@@ -58,282 +73,434 @@ public void setUp() {
+    Reservation expectedReservationMock = mock(Reservation.class);
     ReservationElement reservationElementMock = mock(ReservationElement.class);
-    String reservationId = "reservationId";
-    TupleType tupleType = TupleType.MULTIPLICATION_TRIPLE_GFP;
-    Reservation reservationMock = mock(Reservation.class);
-    TupleChunkMetaDataEntity metaDataMock = mock(TupleChunkMetaDataEntity.class);
 
-    when(reservationElementMock.getTupleChunkId()).thenReturn(chunkId);
-    when(reservationElementMock.getReservedTuples()).thenReturn(consumption);
-    when(tupleChunkMetaDataStorageServiceMock.getTupleChunkData(chunkId)).thenReturn(metaDataMock);
-    when(metaDataMock.getStatus()).thenReturn(ActivationStatus.UNLOCKED);
-    when(metaDataMock.getReservedMarker()).thenReturn(0L);
-    when(metaDataMock.getNumberOfTuples()).thenReturn(consumption);
-    when(metaDataMock.getTupleType()).thenReturn(tupleType);
-    when(reservationMock.getReservationId()).thenReturn(reservationId);
-    when(reservationMock.getTupleType()).thenReturn(tupleType);
-    when(reservationMock.getReservations()).thenReturn(singletonList(reservationElementMock));
-    when(valueOperationsMock.get(testCachePrefix + reservationId)).thenReturn(null);
+    when(valueOperationsMock.get(testCachePrefix + reservationId))
+        .thenReturn(expectedReservationMock);
+    when(expectedReservationMock.getStatus()).thenReturn(ActivationStatus.UNLOCKED);
+    when(expectedReservationMock.getTupleType()).thenReturn(tupleType);
+    when(reservationElementMock.getReservedTuples()).thenReturn(count);
+    when(expectedReservationMock.getReservations())
+        .thenReturn(singletonList(reservationElementMock));
 
-    reservationCachingService.keepReservation(reservationMock);
-
-    verify(valueOperationsMock).set(testCachePrefix + reservationId, reservationMock);
-    verify(tupleChunkMetaDataStorageServiceMock)
-        .updateReservationForTupleChunkData(chunkId, consumption);
-    verify(consumptionCachingServiceMock)
-        .keepConsumption(anyLong(), eq(tupleType), eq(consumption));
+    assertEquals(
+        expectedReservationMock,
+        reservationCachingService.getUnlockedReservation(reservationId, tupleType, count));
   }
 
   @Test
   public void
-      givenReservationWithSameIdInCache_whenKeepReservation_thenThrowCastorServiceException() {
-    String reservationId = "reservationId";
-    Reservation reservationMock = mock(Reservation.class);
-
-    when(reservationMock.getReservationId()).thenReturn(reservationId);
-    when(valueOperationsMock.get(testCachePrefix + reservationId)).thenReturn(reservationMock);
+      givenCachedReservationForIdMismatchType_whenGetUnlockedReservation_thenThrowCastorServiceException() {
+    UUID chunkId = UUID.fromString("b7b010e0-362b-401c-9560-4cf4b2a68139");
+    String reservationId = "testReservationId";
+    TupleType tupleType = INPUT_MASK_GFP;
+    TupleType mismatchedTupleType = MULTIPLICATION_TRIPLE_GFP;
```

Comment:
Used on only once in remainder of body -> redundant

Reply:
see discussion [above](https://github.com/carbynestack/castor/pull/29#discussion_r868054674)